### PR TITLE
Repeat

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: src/playlist-maker.js [hunk 1]
```diff
@@ -248,11 +248,23 @@ const factory = (player, initialList, initialIndex = 0) => {
...
     next() {
 
-      // Make sure we don't go past the end of the playlist.
-      let index = Math.min(playlist.currentIndex_ + 1, list.length - 1);
+      let nextIndex;
 
-      if (index !== playlist.currentIndex_) {
-        return list[playlist.currentItem(index)];
+      // Repeat
+      if (playlist.repeat_ === true) {
```
Comment (gkatsev) [inline src/playlist-maker.js]: can we default `playlist.repeat_` to false and then just do the `if (playlist.repeat_)` here?

Comment (incompl) [inline src/playlist-maker.js]: My thinking was that I want it to not work if someone manually sets something besides `true` for repeat.

Comment (gkatsev) [inline src/playlist-maker.js]: Can we treat anything that isn't a boolean in `repeat` as a getter?

Comment (incompl) [inline src/playlist-maker.js]: What is the goal here? If it's simplification, that doesn't sound like an overall simplification.

Comment (gkatsev) [inline src/playlist-maker.js]: I really hate code like `booleanVariable === true`. We should just use the boolean. If we're worried that the variable gets overridden by a not-boolean, then we should make sure it's always a boolean.

Comment (incompl) [inline src/playlist-maker.js]: Did the API change we talked about IRL to address this.

File: src/playlist-maker.js [hunk 2]
```diff
@@ -281,7 +295,25 @@ const factory = (player, initialList, initialIndex = 0) => {
...
+    /**
+     * Sets `repeat` option, which makes the "next" video of the last video
+     * is the first one.
+     *
+     * @param {Boolean} val
+     */
+    repeat(val) {
+      if (val !== undefined) {
+        if (typeof val !== 'boolean') {
+          videojs.log.error('Invalid value for repeat: "' + val + '"');
```
Comment (gkatsev) [inline src/playlist-maker.js]: `val` should probably be passed as a separate argument to `log.error`, it should end up being printed correctly in the console then and not just `toString`ed.

Comment (incompl) [inline src/playlist-maker.js]: Good idea

**Comment by gkatsev** [comment]
Weird, I wonder why the vjs 6 tests started failing...

File: src/playlist-maker.js [hunk 3]
```diff
@@ -281,7 +295,25 @@ const factory = (player, initialList, initialIndex = 0) => {
     autoadvance(delay) {
       playlist.autoadvance_.delay = delay;
       autoadvance.setup(playlist.player_, delay);
+    },
+
+    /**
+     * Sets `repeat` option, which makes the "next" video of the last video
+     * is the first one.
```
Comment (ldayananda) [inline src/playlist-maker.js]: Suggestion: `"next" video of the last video is the first one.` -> `"next" video of the last video in the playlist be the first video in the playlist`

Comment (incompl) [inline src/playlist-maker.js]: 👍

**Pull request merged**